pkg/model/iam/iam_builder.go

```diff
@@ -244,6 +234,41 @@ func (b *IAMPolicyBuilder) BuildAWSIAMPolicy() (*IAMPolicy, error) {
 	return p, nil
 }
 
+// addEC2Permissions updates the IAM Policy with statements granting tailored
+// access to EC2 resources, depending on the instance role
+func addEC2Permissions(p *IAMPolicy, iamPrefix string, b *IAMPolicyBuilder, wildcard stringorslice.StringOrSlice) {
+	if (b.Role == api.InstanceGroupRoleNode) || (b.Role == api.InstanceGroupRoleMaster) {
+		p.Statement = append(p.Statement, &IAMStatement{
+			Effect:   IAMStatementEffectAllow,
+			Action:   stringorslice.Slice([]string{"ec2:Describe*"}),
+			Resource: wildcard,
+		})
+	}
+
+	if b.Role == api.InstanceGroupRoleMaster {
+		p.Statement = append(p.Statement,
+			&IAMStatement{
+				Effect: IAMStatementEffectAllow,
+				Action: stringorslice.Slice([]string{
+					"ec2:ModifyInstanceAttribute",
+					"ec2:CreateRoute",
+				}),
+				Resource: wildcard,
+			},
+			&IAMStatement{
+				Effect:   IAMStatementEffectAllow,
+				Action:   stringorslice.Slice([]string{"ec2:*"}),
+				Resource: wildcard,
+				Condition: Condition{
+					"StringEquals": map[string]string{
+						"ec2:ResourceTag/KubernetesCluster": b.Cluster.GetName(),
```

Comment:
Yeah I believe we could do on the presence of a tag, I'll see which condition operator fits best.
I noticed that only some resources have the new tag that allows sharing. It only seems to be present on some shared resources, such as VPCs and Subnets. Unless we add the new tag to everything managed by kops, we wouldn't be able to control access to EC2 resources.